### PR TITLE
Register new package XDiff.jl v0.1.0

### PR DESCRIPTION
**Pull request by attobot**
Repository: [dfdx/XDiff.jl](https://github.com/dfdx/XDiff.jl)
Release: [v0.1.0](https://github.com/dfdx/XDiff.jl/releases/tag/v0.1.0)
Travis: [Travis Build Status](https://travis-ci.org/dfdx/XDiff.jl/builds/263443288)
cc: @dfdx

Please make sure that:
- CI passes for supported Julia versions (if applicable).
- Version bounds reflect minimum requirements.

@dfdx This PR will remain open for 24 hours for feedback (which is optional). If you get feedback, please let us know if you are making changes, and we'll merge once you're done.

### CODE REVIEW

File: XDiff/versions/0.1.0/requires [hunk 1]
```diff
@@ -0,0 +1,4 @@
+julia 0.6
+Compat 0.29
+Iterators 0.3.1
```
Comment (tkelman) [inline XDiff/versions/0.1.0/requires]: you may want to switch to IterTools to avoid precompilation issues from the Base.Iterators name collision

Comment (dfdx) [inline XDiff/versions/0.1.0/requires]: Didn't know about the changes, thanks. Will update the release.

File: XDiff/versions/0.1.0/requires [hunk 2]
```diff
@@ -0,0 +1,4 @@
+julia 0.6
+Compat 0.29
```
Comment (tkelman) [inline XDiff/versions/0.1.0/requires]: are you using anything newer than `@compat abstract type` ? (which is 0.17)

I don't see any uses of `fieldcount` which is the only new feature that was first in 0.29

Comment (tkelman) [inline XDiff/versions/0.1.0/requires]: actually if it's 0.6-only, do you need Compat at all right now?

Comment (dfdx) [inline XDiff/versions/0.1.0/requires]: Indeed, fixed it.

**Pull request merged**